src/app/child-dev-project/previous-schools/previous-schools.component.ts

```diff
@@ -77,52 +99,52 @@ export class PreviousSchoolsComponent
     }
 
     this.records = await this.childrenService.getSchoolsWithRelations(id);
+    this.current = this.records.find((record) => record.isActive());
   }
 
   private async initColumnDefinitions() {
     const schools = await this.schoolsService.getSchools().toPromise();
-    const schoolMap = {};
-    schools.forEach((s) => (schoolMap[s.getId()] = s.name));
-
-    this.columns = [
-      {
-        name: "schoolId",
-        label: "School",
-        inputType: ColumnDescriptionInputType.SELECT,
-        selectValues: schools.map((t) => {
-          return { value: t.getId(), label: t.name };
-        }),
-        valueFunction: (entity: ChildSchoolRelation) =>
-          schoolMap[entity["schoolId"]],
-      },
-
-      {
-        name: "schoolClass",
-        label: "Class",
-        inputType: ColumnDescriptionInputType.TEXT,
-      },
-
-      {
-        name: "start",
-        label: "From",
-        inputType: ColumnDescriptionInputType.DATE,
-      },
-      {
-        name: "end",
-        label: "To",
-        inputType: ColumnDescriptionInputType.DATE,
-      },
-      {
-        name: "result",
-        label: "Result",
-        inputType: ColumnDescriptionInputType.NUMBER,
-        valueFunction: (entity: ChildSchoolRelation) =>
+    this.schoolMap = new Map(schools.map((school) => [school.getId(), school]));
+    this.columns = this.config.columns.map((column) =>
+      this.createColumn(column.id, column.label, column.input)
+    );
+  }
+
+  private createColumn(
+    id: string,
+    label: string,
+    type: string
+  ): ColumnDescription {
+    const column: ColumnDescription = {
+      name: id,
+      label: label,
+      inputType: ColumnDescriptionInputType.TEXT,
+    };
+    switch (type) {
+      case "date":
+        column.inputType = ColumnDescriptionInputType.DATE;
+        break;
+      case "school":
+        this.schoolNaming = label;
+        column.inputType = ColumnDescriptionInputType.SELECT;
+        column.selectValues = new Array(...this.schoolMap.values())
+          .sort((s1, s2) => s1.name.localeCompare(s2.name))
+          .map((t) => {
+            return { value: t.getId(), label: t.name };
+          });
+        column.valueFunction = (entity: ChildSchoolRelation) =>
+          this.schoolMap.get(entity["schoolId"]).name;
+        break;
+      case "percentageResult":
```

Comment:
we are adding layers and layers of configuration (special types here, transformed to more general column types ...) :sweat_smile: 

We need to revisit the configuration and table column systems some time and see whether we can generalize, merge and simplify some of this